### DISCUSSION TITLE
Request: Update EspClass with Reset, Restart, getChipId

### DISCUSSION THREAD

**Discussion by EricLauber** [bxparks/UnixHostDuino#83] [Ideas]
The [esp8266/Arduino](https://github.com/esp8266/Arduino) project includes implementations for several functions in `Esp.h` and `Esp.cpp` such as:

-[reset](https://github.com/esp8266/Arduino/blob/1662248b394740d82824f504a021d4dad3d314b6/cores/esp8266/Esp.cpp#L194)
-[restart](https://github.com/esp8266/Arduino/blob/1662248b394740d82824f504a021d4dad3d314b6/cores/esp8266/Esp.cpp#L199)
-[getChipId](https://github.com/esp8266/Arduino/blob/1662248b394740d82824f504a021d4dad3d314b6/cores/esp8266/Esp.cpp#L243)

Code calling these functions will fail to build with EpoxyDuino with the message:

```
error: ‘class EspClass’ has no member named ‘restart’; did you mean ‘timeval EspClass::start’? (not accessible from this context)
```

The same functions build fine and work as intended within the Arduino IDE. Reset and Restart are similar, although Reset may not properly clear out registers. getChipId returns a `uint32_t` that includes the latter half of the ESP MAC address in decimal (such as `ESP5435756`).

My request is to add functions with these names to EpoxyDuino to support building source that calls them. I think it would initially be acceptable to have calls to Reset and Restart basically do nothing - that would still allow projects that use them to build properly. Maybe a step further would be to print to the console that a restart was happening and to end the program/sketch. For getChipId this could return a default value or number like `ESP0000000` in all cases.

**Comment by bxparks** [comment]
Please send me a PR.
There's a long tail of rarely used Arduino functions, macros, and vendor-specific extensions which are not yet implemented in EpoxyDuino. I'm counting on the community to add them as they come across them.

**Reply by EricLauber** [reply]
[Here is the PR](https://github.com/bxparks/EpoxyDuino/pull/84).
